app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -261,3 +304,169 @@ export async function circularRedirect(query: string) {
     throw redirect(circularURL)
   }
 }
+
+/**
+ * Gets all entries in circulars_history for a given circularId
+ * @param circularId
+ * @returns an array of previous versions of a Circular sorted by version
+ */
+export async function getVersions(circularId: number): Promise<Circular[]> {
+  const db = await tables()
+  const result = await db.circulars_history.query({
+    KeyConditionExpression: 'circularId = :circularId',
+    ExpressionAttributeValues: {
+      ':circularId': circularId,
+    },
+  })
+  return result.Items
+}
+
+/**
+ * Creates a set of changes in circulars_change_requests for users
+ * who do not have moderator permissions
+ *
+ * Throws an HTTP error if:
+ *  - The subject is invalid
+ *  - The body is invalid
+ *  - The user is not signed in
+ *
+ * @param circularId
+ * @param body
+ * @param subject
+ * @param request
+ */
+export async function createChangeRequest(
+  circularId: number,
+  body: string,
+  subject: string,
+  request: Request
+) {
+  validateCircular(subject, body)
+  const user = await getUser(request)
+  if (!user)
+    throw new Response('User is not signed in', {
+      status: 403,
+    })
+  const requestor = formatAuthor(user)
+  const db = await tables()
+  await db.circulars_change_requests.put({
+    circularId,
+    body,
+    subject,
+    requestorSub: user.sub,
+    requestor,
+  })
+}
+
+/**
+ * Gets all change requests for a given circular
+ * @param circularId
+ * @returns
+ */
+export async function getChangeRequests(
+  circularId: number
+): Promise<CircularChangeRequest[]> {
+  const db = await tables()
+  return (
+    await db.circulars_change_requests.query({
+      KeyConditionExpression: 'circularId = :circularId',
+      ExpressionAttributeValues: {
+        ':circularId': circularId,
+      },
+      ProjectionExpression: 'circularId, requestor, requestorSub',
+    })
+  ).Items
+}
+
+/**
+ * Verifies the current user and deletes a specific change request.
+ *
+ * Throws an HTTP error if:
+ *  - The current user is not signed in
+ *  - The current user is not the same as the user who requested
+ *    the changed, OR the current user is not a moderator
+ *
+ * @param circularId
+ * @param requestorSub
+ * @param request
+ */
+export async function verifyAndDeleteChangeRequest(
```

Comment:
None of the other exported functions have names that start with `verifyAnd`. Suggest renaming this `verifyAndDeleteChangeRequest` and the internal function `deleteChangeRequestInternal` or `deleteChangeRequestRaw`.